jangjuseong/app.js

```diff
@@ -50,14 +50,148 @@ app.post('/users/signup', async (req, res) => {
         ?, 
         ?, 
         ?
-        );
+      );
     `,
     [name, email, password, profileImage]
   );
 
   res.status(201).json({ message: 'userCreated' });
 });
 
+app.post('/posts', async (req, res) => {
+  const { title, content, postImgUrl, userId } = req.body;
+
+  await mysqlDataSource.query(
+    `INSERT INTO posts (
+      title,
+      content,
+      post_image_url,
+      user_id
+      )
+      VALUES (
+        ?,
+        ?,
+        ?,
+        ?
+      );
+        `,
+    [title, content, postImgUrl, userId]
+  );
+
+  res.status(201).json({ message: 'postCreated' });
+});
+
+app.get('/posts', async (req, res) => {
+  const posts = await mysqlDataSource.query(
+    `SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      p.id AS postingId,
+      p.post_image_url AS postingImageUrl,
+      p.content AS postingContent
+    FROM posts p
+    INNER JOIN users u ON u.id = p.user_id;
+    `
+  );
+
+  res.status(200).json({ posts });
+});
+
+app.get('/posts/user/:userId', async (req, res) => {
+  const { userId } = req.params;
+
+  await mysqlDataSource
+    .query(
+      `SELECT
+    u.id AS userId,
+    u.profile_image AS userProfileImage,
+    pi.post_informations AS postings
+    FROM
+      users u
+    INNER JOIN (
+      SELECT
+        user_id,
+        JSON_ARRAYAGG(
+          JSON_OBJECT (
+            "postingId", id,
+            "postingImageUrl", post_image_url,
+            "postingContent", content
+          )
+        ) AS post_informations
+      FROM
+        posts
+      GROUP BY
+        user_id
+    ) pi 
+      ON pi.user_id = u.id
+    WHERE
+      u.id = ${userId};
+    `
+    )
+    .then((data) => res.status(200).json({ data }));
+});
+
+app.patch('/posts/:postId/user/:userId', async (req, res) => {
+  const { userId, postId } = req.params;
+  const { content } = req.body;
+  await mysqlDataSource.query(
+    `UPDATE posts
+    SET
+      content = ?
+    WHERE 
+      user_id = ${userId} AND id = ${postId}
+    `,
+    [content]
+  );
```

Comment:
`userId`와 `postId`도 `?` 매핑할 수 있도록 수정해주세요!
```suggestion
  await mysqlDataSource.query(
    `UPDATE posts
    SET
      content = ?
    WHERE 
      user_id = ? AND id = ?
    `, [ content, userId, postId ]
  );
```